### ISSUE TITLE
Buzz stigmergy messages resent when lamport are equal.

### ISSUE THREAD

**Issue by xgroleau** [MISTLab/BittyBuzz#13]
I'm not 100% sure of the problem here, but when I have two robots and one puts in the stigmergy every 10 sec, eventually the message goes back and forth between the two robots without stopping. Eating our CPU and bandwidth.

 It's seems the function bbzlamport_isnewer return true if the lamport are equal, causing the same message to be resent over and over again since it assumes it's true.  [Here](https://github.com/MISTLab/BittyBuzz/blob/cb791538d5c199409efef790f7439d2bedfc313f/src/bittybuzz/bbzmsg.c#L124) is the line where it's called and [here](https://github.com/MISTLab/BittyBuzz/blob/cb791538d5c199409efef790f7439d2bedfc313f/src/bittybuzz/bbzmsg.c#L113) is the definition of the function.
 
 When the msg->vs.lamport = 1 and data->timestamp =1, the function returns true, causing the message to be resent while it should do nothing.
 
 Again I'm not sure if this  is the only cause, maybe I misdiagnosed or maybe there are some problem in my code too, I will continue to debug and update here.
 
 
 Here is my buzz code if it can be of any use
 ```js
function init() {
  counter = 0;
  stig = stigmergy.create(1);
  if(id == 1){
    stig.put("leader", 1);
  }
}

# runs every 100ms
function step() { 
  if(counter == 100 and id == 1) {
    var data = stig.get("leader");
    log("Current data in stig: ", data, " size: ", stig.size());
    counter = 0;
  }
  counter = counter + 1;
}
```

**Comment by beltrame** [comment]
When lamport clocks are equal, the vm should call the conflict resolution function (as in the original Buzz). @AnthonyD973 and @Belkharym, do you remember if it is the case here? It might be that the default conflict resolution is incorrect

**Comment by xgroleau** [comment]
After some searching, it seems the circular buffer skips a field for the outputs messages when appending to the cbuff. On the second call of [bbzoutmsg_queue_append_vstig](https://github.com/MISTLab/BittyBuzz/blob/cb791538d5c199409efef790f7439d2bedfc313f/src/bittybuzz/bbzoutmsg.c#L77) (caused by the get in the script) it seems to skip a field in the buffer, it writes at index 2 instead of at index 1. So when I try to get the next data in the cbuf, an uninitialized (at index 1) bbzmsg_t is obtained (since it was written at index 2). 

In other words, the code does

vm->outmsg.buf[0] = vstig query msg;
vm->outmsg.buf[2] = vstig query msg;

While it should do
vm->outmsg.buf[0] = vstig query msg;
vm->outmsg.buf[1] = vstig query msg;
